src/Mvc/Mvc.Core/src/Infrastructure/ActionMethodExecutor.cs

```diff
@@ -19,8 +19,8 @@ internal abstract class ActionMethodExecutor
             new SyncObjectResultExecutor(),
 
             // Executors for async methods
-            new AwaitableResultExecutor(),
             new TaskResultExecutor(),
+            new AwaitableResultExecutor(),
             new TaskOfIActionResultExecutor(),
             new TaskOfActionResultExecutor(),
```

Comment:
I think we should make sure every executor on this list is called by a test, for example I don't see an added check for "TaskOfActionResultExecutor", does this mean we are missing test coverage for it? Or is it shadowed by TaskOfIActionResultExecutor?

Reply:
Good catch @BrennanConroy , I think `TaskOfIActionResultExecutor` had a duplicate test. I adjusted to account for `TaskOfActionResultExecutor`